lib/testsuite/index.js

```diff
@@ -141,13 +142,36 @@ class TestSuite {
     this.context.modulePath = file;
   }
 
+  validateNightwatchInspectorCriteria() {
+    return (
+      this.argv.debug && 
+      Concurrency.isMasterProcess() && 
+      this.settings.desiredCapabilities.browserName === Browser.CHROME
+    );
+  }
+
   async initClient({initialize = true} = {}) {
     const settings = Object.assign({}, this.settings);
 
     this.client = NightwatchClient.create(settings, this.argv);
     if (initialize) {
       await this.client.initialize();
     }
+
+    if (this.validateNightwatchInspectorCriteria()) {
+      this.inspectorServer = new NightwatchInspectorServer(this.client);
+
+      const desiredCapabilities = this.inspectorServer.addExtensionInChromeOption();
+
+      const {globals} = this.settings;
+      this.settings.globals = {
+        ...globals,
+        settings: {
+          ...globals.settings,
+          desiredCapabilities
+        }
+      };
```

Comment:
Why do we need to change globals?

Reply:
Because I need the updated `desiredCapabilities` inside reporter to write the tests.

Reply:
I don't like this too, if something needs to be updated, why not update that specific property?

Reply:
I tried updating only the globals desiredCapabilities, but it doesn't work unless you change the client's desiredCapabilities. I also don't like this too but for now I didn't find other way around.